lib/axon.ex

```diff
@@ -1528,10 +1528,34 @@ defmodule Axon do
   end
 
   @doc """
-  Splits input graph into a container of `n` input graphs
-  along the given axis.
+  Splits input graph into a container of `m` other input graphs 
+  In case `n` is a number the input graph is split in n slices with the same size
+  In case `n` is a list the input graph is split in Enum.count slices with the 
+  size of the i-th element of the list.
   """
-  def split(%Axon{output_shape: shape} = parent, n, opts \\ []) do
+
+  def split(parent, n, opts \\ []) 
+
+  def split(%Axon{output_shape: shape} = parent, n, opts) when is_list(n) do
+    axis = opts[:axis] || -1
+    names = opts[:name] || []
```

Comment:
I wonder if for output names we default to something like `input_name_split_n` where `n` is the corresponding split and input name is the parent's name.

Reply:
I was thinking to a default layer name but couldn't decide on something good. It is OK for me your proposal.

Reply:
How can I get the name of the parent layer? I noticed that now `parent.name` is returning a function expecting op and op_counts parameters, but I cannot understand what I must put into those two variables.